### DISCUSSION TITLE
WebMap onto Form View

### DISCUSSION THREAD

**Discussion by AdminABT** [gityopie/odoo-addons#191] [Q&A]
Hi @gityopie 
Nice development on this Odoo / Google integration
 I am trying to use the web_google_maps add-on in the Odoo 15 Enterprise version in order to deploy the Map under a Form view for a custom object.
As an example, I have a Car object with his Lat/Lon as attributes and am trying to have a way to display where the car is within its form view
Somehow I ma struggeling to have it working. Would you mind confirming if such is achievable with your baseline ?

**Comment by gityopie** [comment]
Hi @AdminABT 

One possible way is to define `One2many` fields that store the geolocation record, this way you can load `google_map` view inside `form` view.
For example:

```python
class TheCarGeolocation(models.Model):
    _name = 'the.car.geolocation'

    latitude = fields.Float()
    longitude = fields.Float()
    thecar_id = fields.Many2one('the.car', required=True)


class TheCar(models.Model):
    _name = 'the.car'

    car_geolocation = fields.One2many('the.car.geolocation', 'thecar_id')
```

Some where inside `TheCar` form view
```xml
<field name="car_geolocation" mode="google_map"/>
```

I know this solution is not the best one but it works

I will try to check, maybe it's a new widget.    
Thanks for your feedback!

**Reply by AdminABT** [reply]
HI @gityopie 
Thanks a lot for your response, I gave it a try but I should have miss something on how this view work....
Getting the model, submodel for the coordinates and so on but each time I am trying to display the coordinates under a mode="google_map", I am getting an ... Uncaught Promise > Renderer is not a constructor

**Reply by gityopie** [reply]
@AdminABT have you try to create `google_map` view for the submodel?

**Reply by AdminABT** [reply]
@gityopie I probably (surely) did it wrong but yes I had to try to created one

**Reply by AdminABT** [reply]
@gityopie Just to give you an update, I succeed to get a differnet behaviour, the above issue was due to the missnig template on the view. 
Now I can properly see the Map view when I am on the TheCarGeolocation object 
unfortunately, this does not popup inside TheCar form view - just see the name of the Geolocaiton but the map does not render

**Reply by gityopie** [reply]
@AdminABT I just pushed a new branch [15.0-google_map_inline](https://github.com/gityopie/odoo-addons/tree/15.0-google_map_inline) to address issue when `google_map` view used nested by `one2many` fields inside `form` view.

I tried your use case and this is how I did it
```python
class PartnerGeolocation(models.Model):
    _name = 'partner.geolocation'
    _description = 'Partner geolocation'

    partner_id = fields.Many2one(
        'res.partner', string='Partner', required=True
    )
    name = fields.Char(related='partner_id.name', string='Name')
    latitude = fields.Float(string='Latitude', required=True)
    longitude = fields.Float(string='Longitude', required=True)


class ResPartner(models.Model):
    _inherit = 'res.partner'

    partner_geolocation_ids = fields.One2many(
        'partner.geolocation', 'partner_id', string='Partner Geolocation'
    )
```

```xml
<record id="view_partner_form_contacts_google_map_inherit_coba" model="ir.ui.view">
    <field name="name">view.partner.form.contacts.google.map.inherit.coba</field>
    <field name="model">res.partner</field>
    <field name="priority">5000</field>
    <field name="inherit_id" ref="base.view_partner_form"/>
    <field name="arch" type="xml">
        <xpath expr="//notebook" position="before">
            <field name="partner_geolocation_ids" mode="google_map" context="{'default_partner_id': active_id}">
                <form string="Locations">
                    <sheet>
                        <group>
                            <group>
                                <field name="latitude"/>
                                <field name="longitude"/>
                                <field name="partner_id" invisible="1"/>
                            </group>
                        </group>
                    </sheet>
                </form>
                <google_map string="Locations" lat="latitude" lng="longitude" color="green">
                    <field name="latitude"/>
                    <field name="longitude"/>
                    <field name="partner_id"/>
                    <field name="name"/>
                    <templates>
                        <t t-name="kanban-box">
                            <div class="oe_kanban_details">
                                <field name="name"/>
                                <field name="partner_id"/>
                                <field name="latitude"/>
                                <field name="longitude"/>
                            </div>
                        </t>
                    </templates>
                </google_map>
            </field>
        </xpath>
    </field>
</record>
```
The result looks like this
<img width="766" alt="Screen Shot 2022-04-06 at 12 47 32" src="https://user-images.githubusercontent.com/3539144/161904287-2ab2cc1e-5b79-4e0e-bdd6-c34e95a3419d.png">

**Reply by AdminABT** [reply]
@gityopie That's Awsome. Works very well
Thanks a lot
